htdocs/fourn/facture/paiement.php

```diff
@@ -171,7 +172,7 @@
 	        } elseif (substr($key, 0, 21) == 'multicurrency_amount_')
 			{
 				$cursorfacid = substr($key, 21);
-	            $multicurrency_amounts[$cursorfacid] = price2num(trim(GETPOST($key)));
+	            $multicurrency_amounts[$cursorfacid] = GETPOST($key) ? price2num(trim(GETPOST($key))) : 0;
```

Comment:
Instead of 
= GETPOST($key) ? ... , you probably want to do 
= GETPOSTISSET($key) ? ...
Right ?

Reply:
Hello,

when I use
```
$multicurrency_amounts[$cursorfacid] = GETPOSTISSET($key) ? price2num(trim(GETPOST($key))) : 0;
```
the error is still present, only with
```
$multicurrency_amounts[$cursorfacid] = GETPOST($key) ? price2num(trim(GETPOST($key))) : 0;
```
it is working.

I have currently no idea why

Reply:
a little bit more investigation
```
// htdocs\core\lib\functions.lib.php - line 271
		$isset = (isset($_POST[$paramname]) || isset($_GET[$paramname]));
```
when i debug with
```
	print "<br>POST:".$_POST[$paramname]." -> ".isset($_POST[$paramname]);
	print "<br>GET:".$_GET[$paramname]." -> ".isset($_GET[$paramname]);
```

i became
```
POST: -> 1
GET: -> 

Warning: A non-numeric value encountered in C:\wamp64\www\dolibarr\htdocs\fourn\facture\paiement.php on line 178
```

Reply:
https://stackoverflow.com/questions/13045279/if-isset-post

Answer 3

> Use !empty instead of isset. isset return true for $_POST because $_POST array is superglobal and always exists (set).
> Or better use $_SERVER['REQUEST_METHOD'] == 'POST'